src/Types/ReceiveName.cs

```diff
@@ -76,4 +100,73 @@ private static bool IsValid(string name, out ValidationError? error)
         error = null;
         return true;
     }
+
+    /// <summary>
+    /// Attempt to deserialize a span of bytes into a smart contract receive name.
+    /// </summary>
+    /// <param name="bytes">The span of bytes potentially containing a receive name.</param>
+    /// <param name="output">Where to write the result of the operation</param>
+    public static bool TryDeserial(ReadOnlySpan<byte> bytes, out (ReceiveName? receiveName, string? Error) output)
+    {
+        if (bytes.Length < MinSerializedLength)
+        {
+            var msg = $"Invalid length of input in `ReceiveName.TryDeserial`. Expected at least {MinSerializedLength}, found {bytes.Length}";
+            output = (null, msg);
+            return false;
+        };
+        var sizeRead = BinaryPrimitives.ReadUInt16BigEndian(bytes); // This should never throw, since we already checked the length.
+        var size = sizeRead + MinSerializedLength;
```

Comment:
Like for parameters, I think this is really an odd way of conceptualizing it. YOu hardcode UInt16 just in the line above, but then you have this constant MinSerializedLength for reasons I do not understand.